test/aws/rails/sqs_active_job/configuration_test.rb

```diff
@@ -36,6 +36,16 @@ module SqsActiveJob
           expect(cfg.to_h).to include(expected)
         end
 
+        it 'accepts YAML config with alias' do
+          allow_any_instance_of(ERB).to receive(:result).and_return(<<~YAML)
+common: &common
+  default: 'https://queue-url'
+queues:
+  <<: *common
+          YAML
+          expect { Aws::Rails::SqsActiveJob::Configuration.new }.to_not raise_error
```

Comment:
If `YAML.load` for such a YAML runs on Ruby 3.1.0, it raises `Psych::BadAlias`. This is the breaking change. (Not only for alias though)

```
Failure:
Aws::Rails::SqsActiveJob::Configuration#test_0005_merges runtime options with YAML options [/Users/m-ohba/ghq/github.com/ohbarye/aws-sdk-rails/gemfiles/vendor/bundle/ruby/3.1.0/gems/rspec-expectations-3.12.0/lib/rspec/expectations/handler.rb:40]:
expected no Exception, got #<Psych::BadAlias: Unknown alias: common> with backtrace:
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:430:in `visit_Psych_Nodes_Alias'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each_slice'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `each_slice'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:343:in `revive_hash'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:318:in `visit_Psych_Nodes_Document'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:30:in `visit'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/visitor.rb:6:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/visitors/to_ruby.rb:35:in `accept'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych.rb:335:in `safe_load'
  # /Users/m-ohba/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych.rb:370:in `load'
  # /Users/m-ohba/ghq/github.com/ohbarye/aws-sdk-rails/lib/aws/rails/sqs_active_job/configuration.rb:171:in `load_yaml'
```

Reply:
This makes sense. Could we add a comment in code to this test definition that briefly references this issue OR explains why we need to test it?